presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -2054,6 +2054,30 @@ public void testJoinWithConstantExpression()
     }
 
     @Test
+    public void testJoinWithConstantExpressionWithCoercion()
+    {
+        // Covers #7520
+        {
```

Comment:
These nested scopes look strange. I'd either make them two separate tests or reuse the variables.